core/src/jsMain/kotlin/Peripheral.kt

```diff
@@ -125,28 +125,28 @@ public class JsPeripheral internal constructor(
     private val gatt: BluetoothRemoteGATTServer
         get() = bluetoothDevice.gatt!! // fixme: !!
 
-    private val connectJob: SharedRepeatableTask = scope.sharedRepeatableTask {
+    private val connectJob: SharedRepeatableTask<Unit> = scope.sharedRepeatableTask {
         try {
             openConnection()
         } catch (cause: Throwable) {
             logger.error(cause) { message = "Failed to connect" }
-            disconnectJob.launch()
+            @Suppress("DeferredResultUnused") // Safe to ignore Deferred because the result is shared elsewhere
+            disconnectJob.getOrAsync()
             throw IOException("Connection attempt failed", cause)
         }
     }
 
-    private val disconnectJob: SharedRepeatableTask = scope.sharedRepeatableTask {
-        connectJob.join()
+    private val disconnectJob: SharedRepeatableTask<Unit> = scope.sharedRepeatableTask {
+        connectJob.cancelAndJoin()
         closeConnection()
     }
 
     public override suspend fun connect() {
-        connectJob.launch().join()
+        connectJob.getOrAsync().await()
     }
 
     public override suspend fun disconnect() {
-        connectJob.cancelAndJoin()
-        disconnectJob.launch().join()
+        disconnectJob.getOrAsync().await()
```

Comment:
Again, not-obvious and no comments, my bad.

Strongly suggest the `connectJob.cancelAndJoin()` remain here and move that change to a separate PR.